core/pyos.py

```diff
@@ -266,26 +261,56 @@ def InputAvailable(fd):
 
 UNTRAPPED_SIGWINCH = -1
 
-class SigwinchHandler(object):
-  """Wrapper to call user handler."""
 
-  def __init__(self, display, sig_state):
-    # type: (_IDisplay, SignalState) -> None
-    self.display = display
-    self.sig_state = sig_state
-    self.user_handler = None  # type: _TrapHandler
+class _SignalHandler(object):
+  """
+  A singleton that implements a basic generic signal handler that enqueues
+  signals for asynchrnous processing as they are fired.
+  """
+  _instance = None  # type: _SignalHandler
 
-  def __call__(self, sig_num, unused_frame):
-    # type: (int, Any) -> None
-    """For Python's signal module."""
+  signal_queue = []  # type: List[int]
+  last_sig_num = 0  # type: int
+  sigwinch_num = UNTRAPPED_SIGWINCH
+
+  def __new__(cls):
+    # type: () -> _SignalHandler
+    if cls._instance is None:
+        cls._instance = super(_SignalHandler, cls).__new__(cls)
 
-    # SENTINEL for UNTRAPPED SIGWINCH.  If it's trapped, self.user_handler
-    # will overwrite it with signal.SIGWINCH.
-    self.sig_state.last_sig_num = UNTRAPPED_SIGWINCH
+    return cls._instance
```

Comment:
Can we just create a global like:

```
gSignalHandler = SignalHandler()

```

and then 

```
def LastSignal():
  return gSignalHander.LastSignal()
```

This is just a style thing but I think it makes the  code more consistent